enterprise/backend/src/metabase_enterprise/ai_sql_fixer/api.clj

```diff
@@ -20,9 +19,9 @@
                                      [:error_message :string]]]
   (binding [persisted-info/*allow-persisted-substitution* false]
     (qp.perms/check-current-user-has-adhoc-native-query-perms query)
-    (let [driver (driver.u/database->driver (:database query))
-          compiled (qp.compile/compile-with-inline-parameters query)]
-      (-> (metabot-v3.client/fix-sql {:sql (:query compiled)
+    (let [sql    (-> query :native :query)
+          driver (driver.u/database->driver (:database query))]
+      (-> (metabot-v3.client/fix-sql {:sql sql
```

Comment:
@ranquild , are the `"template-tags"` mentioned in #53117 relevant? If yes, what role do/should they play?

Reply:
Nope, not relevant. We just send what we get with `Lib.toLegacyQuery`. But template tags aren't used at all

Reply:
OK, I'll update the epic then.